src/bindings/python/flux/uri/resolvers/lsf.py

```diff
@@ -0,0 +1,91 @@
+###############################################################
+# Copyright 2022 Lawrence Livermore National Security, LLC
+# (c.f. AUTHORS, NOTICE.LLNS, COPYING)
+#
+# This file is part of the Flux resource manager framework.
+# For details, see https://github.com/flux-framework.
+#
+# SPDX-License-Identifier: LGPL-3.0
+###############################################################
+
+"""
+This module supports the Flux URI resolver for IBM Spectrum LSF.
+
+It can be used like:
+  `flux uri --remote lsf:<JOBID>`.
+
+"""
+
+import getpass
+import os
+import subprocess
+from pathlib import PurePath
+
+import yaml
+
+import flux.hostlist as hostlist
+from flux.uri import URIResolverPlugin
+
+
+def lsf_find_compute_node(jobid):
+    """Figure out where the job is being run using YAML output from IBM Cluster Systems Manager."""
+
+    csm_path = os.getenv(
+        "CSM_ALLOCATION_QUERY", "/opt/ibm/csm/bin/csm_allocation_query"
```

Comment:
Since we can override `csm_allocation_query` now, we could add basic sanity tests to `make check` by setting `CSM_ALLOCATION_QUERY` to a script which returns known output and ensuring the "LSF" jobid resolves to what is expected. (you may also have to add a fake `ssh` script, there are examples of this in the testsuite).

This is what is done for the `slurm` uri resolver  in `t2802-uri-cmd.t`.